React/Views/RCTPicker.m

```diff
@@ -70,6 +70,10 @@ - (NSString *)pickerView:(__unused UIPickerView *)pickerView
   return [RCTConvert NSString:_items[row][@"label"]];
 }
 
+- (CGFloat)pickerView:(UIPickerView *)pickerView rowHeightForComponent:(NSInteger)component {
+    return _font.pointSize + 8;
```

Comment:
`_font` is picker property, not item property, so it is unrelated to items styling at all.